smithy-typescript-ssdk-libs/server-apigateway/src/lambda.ts

```diff
@@ -42,14 +68,69 @@ export function convertResponse(response: HttpResponse): APIGatewayProxyResultV2
   };
 }
 
+export function convertVersion1Response(response: HttpResponse): APIGatewayProxyResult {
+  return {
+    statusCode: response.statusCode,
+    multiValueHeaders: convertResponseHeaders(response.headers),
+    body: response.body,
+    isBase64Encoded: false,
+  };
+}
+function convertResponseHeaders(headers: HeaderBag) {
+  const retVal: { [key: string]: string[] } = {};
+  for (const [key, val] of Object.entries(headers)) {
+    retVal[key] = val.split(",").map((v) => v.trim());
+  }
+  return retVal;
+}
+
+function isV2Event(event: APIGatewayProxyEvent | APIGatewayProxyEventV2): event is APIGatewayProxyEventV2 {
+  return hasVersion(event) && event.version === "2.0";
+}
+
+function hasVersion(event: any): event is Record<"version", string> {
+  return event.hasOwnProperty("version");
+}
+
+function convertMultiValueHeaders(multiValueHeaders: APIGatewayProxyEventMultiValueHeaders | null) {
```

Comment:
seems like this could still return `: HeaderBag`?

Reply:
Yeah, it could. It's a private function with an inferred return type, don't know how much it matters to be explicit.

Reply:
I mentioned only because `convertHeaders` and some other private functions have it.